vllm/model_executor/layers/fused_moe/cutlass_moe.py

```diff
@@ -480,3 +480,110 @@ def cutlass_moe_fp4(a: torch.Tensor, a1_gscale: torch.Tensor,
     out = (c2.view(m, num_topk, k) *
            topk_weights.view(m, num_topk, 1).half()).sum(dim=1)
     return out.to(dtype=out_dtype)
+
+
+def _valid_cutlass_block_scaled_grouped_gemm(hidden_states: torch.Tensor, w1: torch.Tensor, w2: torch.Tensor) -> bool:
+
+    def _valid_cutlass_block_scaled_grouped_gemm_shape(M: int, N: int, K: int):
+        return 128 <= M and N % 128 == 0 and K % 128 == 0
+
+    m = hidden_states.size(0)
+    _, K, N = w2.size()
+    if not _valid_cutlass_block_scaled_grouped_gemm_shape(m, N, K):
+        logger.debug("CutlassBlockScaledGroupedGemm disabled: unalinged problem size.")
+        return False
+    
+    if (w1.dtype != torch.float8_e4m3fn or w2.dtype != torch.float8_e4m3fn):
+        logger.debug("CutlassBlockScaledGroupedGemm disabled: invalid weight dtype(s).")
+        return False
+    
+    return True
+
+
+def run_cutlass_block_scaled_fused_experts(
```

Comment:
Can you also integrate this with `triton_deep_gemm_moe.py` (maybe we should change this name) so it can be used with EP?